examples/foreign_call_on_tick/main_test.go

```diff
@@ -18,21 +19,46 @@ import (
 )
 
 func TestPluginContext_OnTick(t *testing.T) {
-	opt := proxytest.NewEmulatorOption().WithVMContext(&vmContext{})
-	host, reset := proxytest.NewHostEmulator(opt)
-	defer reset()
-
-	// Call OnVMStart.
-	require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
-	require.Equal(t, tickMilliseconds, host.GetTickPeriod())
-
-	// Register foreign function named "compress".
-	host.RegisterForeignFunction("compress", func(b []byte) []byte { return b })
-
-	for i := 1; i < 10; i++ {
-		host.Tick()
-		// Check Envoy logs.
-		logs := host.GetInfoLogs()
-		require.Contains(t, logs, fmt.Sprintf("foreign function (compress) called: %d, result: %s", i, "68656c6c6f20776f726c6421"))
-	}
+	vmTest(t, func(t *testing.T, vm types.VMContext) {
+		opt := proxytest.NewEmulatorOption().WithVMContext(vm)
+		host, reset := proxytest.NewHostEmulator(opt)
+		defer reset()
+
+		// Call OnVMStart.
+		require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
+		require.Equal(t, tickMilliseconds, host.GetTickPeriod())
+
+		// Register foreign function named "compress".
+		host.RegisterForeignFunction("compress", func(b []byte) []byte { return b })
+
+		for i := 1; i < 10; i++ {
+			host.Tick()
+			// Check Envoy logs.
+			logs := host.GetInfoLogs()
+			require.Contains(t, logs, fmt.Sprintf("foreign function (compress) called: %d, result: %s", i, "68656c6c6f20776f726c6421"))
+		}
+	})
+}
+
+// vmTest executes f twice, once with a types.VMContext that executes plugin code directly
+// in the host, and again by executing the plugin code within the compiled main.wasm binary.
+// Execution with main.wasm will be skipped if the file cannot be found, allowing tests to run
+// with Go even if TinyGo isn't present.
```

Comment:
Everyone is required to have TinyGo as you see in the README. Let's remove this commentary.

```suggestion
// Execution with main.wasm will be skipped if the file cannot be found.
```